app/containers/WriteReviewPage/index.js

```diff
@@ -9,33 +9,69 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { createStructuredSelector } from 'reselect';
 import { compose } from 'redux';
-import Select from 'react-select';
-import Header from 'components/Header';
-import PageSection from 'components/PageSection';
-import BorderBottomDiv from 'components/shared/BorderBottomDiv';
-import Stars from 'components/Stars';
 import injectReducer from 'utils/injectReducer';
 import injectSaga from 'utils/injectSaga';
 
-import 'react-select/dist/react-select.css';
-
-import { locationList, selectedLocationChange } from './actions';
-import { selectLocationList, selectSelectedLocationId } from './selectors';
+import { locationList, guideList, selectedLocationChange } from './actions';
+import { selectLocationList, selectGuideList, selectSelectedLocationId } from './selectors';
 import reducer from './reducer';
 import saga from './saga';
 
+import Helmet from 'react-helmet'
+
+import Header from 'components/Header';
+import PageSection from 'components/PageSection';
+import BorderBottomDiv from 'components/shared/BorderBottomDiv';
+import Select from 'components/Select';
+import Stars from 'components/Stars';
+import GuideReviewForm from 'components/GuideReviewForm';
+
 export class WriteReviewPage extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
 
   constructor(props) {
     super(props);
-
+    this.handleDetailsChange = this.handleDetailsChange.bind(this);
+    this.handleConfirmGuided = this.handleConfirmGuided.bind(this);
+    this.handleSelectGuideId = this.handleSelectGuideId.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
+
+    // State should contain form data for easy validation and submit.
+    this.state = {
+      // Should figure out a way to pass selectedId props based on a review page we're on
+      selectedLocationId: props.selectedLocationId ? props.selectedLocationId : null,
+      tripRating: 0,
+      details: '',
+      selectedGuideId: props.selectedGuideId ? props.selectedGuideId : null,
+      guideRating: 0,
+    }
   }
 
   componentDidMount() {
+    // API call
     this.props.requestLocationList();
   }
 
+  handleDetailsChange(e) {
+    this.setState({ details: e.target.value });
+  }
+
+  // This function is passed to GuideReviewForm component
+  handleConfirmGuided() {
+    // API call
+    this.props.requestGuideList();
+  }
+
+  // This function is passed to GuideReviewForm component
+  handleSelectGuideId(guide) {
+    if(guide) {
+    console.log(guide);
+      this.setState({ selectedGuideId: guide.value});
+    }
+    else {
+      this.setState({ selectedGuideId: null });
```

Comment:
Sorry, but this is wrong. Since we're doing redux we should never touch state directly, only in reducers. I think there's a bug here anyway, it overwrites all state. This is the point of redux, to protect states, isolate code paths that write to state. 

All these should be action, reducer pairs. I'll review in more depth later but this was glaring

Reply:
I was researching this and came across an article/blog (from Aug. 2016) which states:

"The current best practice is to use local state to handle the state of your user interface (UI) state rather than data. For example, using a controlled component to fill out a form is a perfectly valid use of local state."

This seems really sensible to me, and I would think it would keep the code from being bloated by reaching for the Redux store for the simplest interactions (which also might have the effect of updating React trees more than needed for tiny prop changes?). Here's a link to what I read: https://medium.freecodecamp.org/where-do-i-belong-a-guide-to-saving-react-component-data-in-state-store-static-and-this-c49b335e2a00.

Reply:
Thanks for the link, that's a helpful read. I do think it is a bit out of date, looks like a lot of people are using redux-form and such to manage things like this under the hood

This is actually super important to decide, set a standard on. Much of our app will be forms in one way or another, simple scalable code is crucial here. I'm not married to throwing everything in redux, but I do think we need to be smart about what we pick here. Let me be clear, the code i've written thus far is definitely not 100% correct in this sense. But we should def have a standard.

Going to talk to some friends of mine about best practices here. Thanks for the research, this is a good push

Reply:
I definitely agree there should be a standard. I'm very interested in knowing what's considered best practice (though I know it can differ between developers). I consider you to be the one in charge here, though, so I'll go with what you think is best.

Reply:
👍 Give me a day or so and I'll get back to you here. For now I think it's fine as it is. But we should settle on a strong standard